metrics_api/lib/opentelemetry/metrics/meter.rb

```diff
@@ -8,71 +8,162 @@ module OpenTelemetry
+      # @param name [String]
+      #   Must conform to the instrument name syntax:
+      #   not nil or empty, case-insensitive ASCII string that matches `/\A[a-z][a-z0-9_.-]{0,62}\z/i`
+      # @param unit [optional String]
+      #   Must conform to the instrument unit rule:
+      #   case-sensitive ASCII string with maximum length of 63 characters
+      # @param description [optional String]
+      #   Must conform to the instrument description rule:
+      #   UTF-8 string but up to 3 bytes per charater with maximum length of 1023 characters
+      # @param advice [optional Hash] Set of recommendations aimed at assisting
+      #   implementations in providing useful output with minimal configuration
+      #
+      # @return [Instrument::Counter]
+      def create_counter(name, unit: nil, description: nil, advice: nil)
+        create_instrument(:counter, name, unit, description, advice, nil) { NOOP_COUNTER }
       end
 
-      def create_histogram(name, unit: nil, description: nil)
-        create_instrument(:histogram, name, unit, description, nil) { HISTOGRAM }
+      # @param name [String]
+      #   Must conform to the instrument name syntax:
+      #   not nil or empty, case-insensitive ASCII string that matches `/\A[a-z][a-z0-9_.-]{0,62}\z/i`
+      # @param unit [optional String]
+      #   Must conform to the instrument unit rule:
+      #   case-sensitive ASCII string with maximum length of 63 characters
+      # @param description [optional String]
+      #   Must conform to the instrument description rule:
+      #   UTF-8 string but up to 3 bytes per charater with maximum length of 1023 characters
+      # @param advice [optional Hash] Set of recommendations aimed at assisting
+      #   implementations in providing useful output with minimal configuration
+      #
+      # @return [Instrument::Histogram]
+      def create_histogram(name, unit: nil, description: nil, advice: nil)
+        create_instrument(:histogram, name, unit, description, advice, nil) { NOOP_HISTOGRAM }
       end
 
-      def create_up_down_counter(name, unit: nil, description: nil)
-        create_instrument(:up_down_counter, name, unit, description, nil) { UP_DOWN_COUNTER }
+      # @param name [String]
+      #   Must conform to the instrument name syntax:
+      #   not nil or empty, case-insensitive ASCII string that matches `/\A[a-z][a-z0-9_.-]{0,62}\z/i`
+      # @param unit [optional String]
+      #   Must conform to the instrument unit rule:
+      #   case-sensitive ASCII string with maximum length of 63 characters
+      # @param description [optional String]
+      #   Must conform to the instrument description rule:
+      #   UTF-8 string but up to 3 bytes per charater with maximum length of 1023 characters
+      # @param advice [optional Hash] Set of recommendations aimed at assisting
+      #   implementations in providing useful output with minimal configuration
```

Comment:
It seems `advice` has been added to the specification recently. I'm not sure how it will be exactly used. For now, I'm thinking of it as an "options" Hash with recommended values. For example in docs, [the recommended bucket boundaries for a histogram](https://opentelemetry.io/docs/specs/otel/metrics/api/#instrument-advice). I _imagine_ they will define standard names to be used in `advice`, maybe:

```rb
advice: {
  histogram: {
    explicit_bucket_boundaries: [0.1, 0.5, 1.0, 5.0, 10.0, 25.0, Float::INFINITY]
  }
}
```

Reply:
Reading the SDK page, I found another reference to `advice` in [Default Aggregation](https://opentelemetry.io/docs/specs/otel/metrics/sdk/#default-aggregation):

> The Default Aggregation informs the SDK to use the Instrument `kind` to select an aggregation and `advice` to influence aggregation configuration parameters (as noted in the "Selected Aggregation" column).

> Instrument Kind | Selected Aggregation
>  -- | --
>  [...] | [...]
>  Histogram | Explicit Bucket Histogram Aggregation, with `ExplicitBucketBoundaries` from `advice` if provided

Reply:
I don't think this should block the current PR from being merged, but it seems that [`advice` has been renamed to `advisory parameters`](https://github.com/open-telemetry/opentelemetry-specification/pull/3662). 

* API: https://opentelemetry.io/docs/specs/otel/metrics/api/#instrument-advisory-parameters 
* SDK: https://opentelemetry.io/docs/specs/otel/metrics/sdk/#instrument-advisory-parameters 

It has a mixed status in the API and is experimental in the SDK.